src/components/molecules/Button/Button.tsx

```diff
@@ -0,0 +1,77 @@
+import { FC, PropsWithChildren, SVGProps, FunctionComponent } from 'react'
+import BaseButton, {
+  BaseButtonProps,
+} from 'components/atoms/BaseButton/BaseButton'
+import classNames from 'classnames'
+import classes from './styles.module.scss'
+
+export enum AppearanceTypes {
+  Primary = 'primary',
+  Secondary = 'secondary',
+  Text = 'text',
+}
+
+export enum SizeTypes {
+  M = 'm',
+  S = 's',
+}
+
+export interface ButtonProps extends BaseButtonProps {
+  appearance?: AppearanceTypes
+  size?: SizeTypes
+  icon?: FunctionComponent<SVGProps<SVGSVGElement>>
+  ariaLabel?: string
+  hidden?: boolean
+  disabled?: boolean
+  className?: string
+}
+
+export type IconProps = {
+  icon?: FunctionComponent<SVGProps<SVGSVGElement>>
+  className?: string
+  ariaLabel?: string
+}
+
+const Icon: FC<IconProps> = ({ icon: IconComponent, className, ariaLabel }) => {
+  const ariaLabelToUse = ariaLabel
+
+  if (!IconComponent) return null
+  return (
```

Comment:
**[blocker]**

Some issues with the icon:

1. Color of the icon seems to be wrong, at least in secondary button.
2. Seems that the icon is in front of the text for secondary button and after the text in primary. We should have the capability to have the icon on either side of the text for any button, but this also means that by default we should always choose 1 side (either before or after the text).